src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala

```diff
@@ -67,33 +92,45 @@ case object ThurloeDatabaseConnector extends DataAccess with LazyLogging {
     } yield result
   }
 
-  def lookupIncludingDatabaseId(userId: String, key: String): Future[UserKeyValuePairWithId] =
+  def lookupIncludingDatabaseId(userId: String, key: String)(implicit samDAO: SamDAO): Future[UserKeyValuePairWithId] =
     for {
-      results <- lookupWithConstraint(x => x.key === key && x.userId === userId)
+      samUser <- lookupSamUser(userId)
+      results <- lookupWithConstraint(x =>
+        x.key === key && (x.userId === samUser.getId || x.userId === samUser.getGoogleSubjectId || x.userId === samUser.getAzureB2CId)
+      )
       result <- if (results.isEmpty) {
         Future.failed(new KeyNotFoundException(userId, key))
       } else if (results.size == 1) {
         Future.successful(results.head)
       } else {
         Future.failed(InvalidDatabaseStateException(s"Too many results: ${results.size}"))
       }
-    } yield result
+    } yield result.copy(userKeyValuePair = result.userKeyValuePair.copy(userId = userId))
```

Comment:
this is so the id used in the request is always the one returned with the results object